torch/_weights_only_unpickler.py

```diff
@@ -151,8 +176,31 @@ def load(self):
                 full_path = f"{module}.{name}"
                 if full_path in _get_allowed_globals():
                     self.append(_get_allowed_globals()[full_path])
+                elif full_path in _get_user_allowed_globals():
+                    self.append(_get_user_allowed_globals()[full_path])
                 else:
-                    raise RuntimeError(f"Unsupported class {full_path}")
+                    class_type = getattr(modules[module], name)
```

Comment:
@albanD I think there might be a fundamental problem here that prevents us from allowlisting *all* tensor subclasses.

The `GLOBAL <bla>` in the pickle file will have <bla> as a string, in order to check `issubclass(bla, torch.Tensor), we need to convert the string `bla` to the actual type object. To my knowledge, I could not find a safe way to do this, and I suspect this line here might not be what we consider secure. wdyt?

Is there a way where we can constrain this such that it is still secure?